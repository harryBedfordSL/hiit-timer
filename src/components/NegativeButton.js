import React, { Component } from 'react';
import styles from './NegativeButton.module.css';
import { Icon } from 'semantic-ui-react';

export default class NegativeButton extends Component {    
    render() {
        return (
            <button
                className={styles.negBtn}
                onClick={this.props.onClick}
            >
                <Icon
                    className={styles.icon}
                    name={this.props.icon} 
                    size='small'
                />
            </button>
        )
    }
}