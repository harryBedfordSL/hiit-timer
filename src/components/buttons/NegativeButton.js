import React from 'react';
import styles from './NegativeButton.module.css';
import { Icon } from 'semantic-ui-react';
import { withRedux } from '../../utils/WithRedux';

export default withRedux(function NegativeButton({ onClick, icon, size, theme}) {    
    return (
        <button
            style={theme.negativeButton}
            className={styles.negBtn}
            onClick={onClick}
        >
            <Icon
                className={styles.icon}
                name={icon} 
                size={size}
            />
        </button>
    )
})