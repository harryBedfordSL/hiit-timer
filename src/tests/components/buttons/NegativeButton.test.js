import React from 'react';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';
import NegativeButton from '../../../components/buttons/NegativeButton';

// move arranges to standardised place
// const onClick = () => {};
// const icon = 'close';
// const size = 'small';
// const theme = 'test';

describe('Negative button tests', () => {
    // let btn;
    // beforeEach(() => {
    //     negativeButton = shallow(<NegativeButton
    //         onClick={onClick} 
    //         icon={icon}
    //         size={size}
    //         theme={theme}
    //     />)
    // })

    test('renders without crashing (with snapshot)', () => {
        // Arrange
        const onClick = () => {};
        const icon = 'close';
        const size = 'small';
        const theme = 'test';
        
        // Act
        const negBtn = create(<NegativeButton 
            onClick={onClick} 
            icon={icon}
            size={size}
            theme={theme}
        />);

        // Assert
        expect(negBtn.toJSON()).toMatchSnapshot();
    })

    test('renders without crashing (with enzyme)', () => {
        // Arrange
        const onClick = () => {};
        const icon = 'close';
        const size = 'small';
        const theme = 'test';
        
        // Act
        const wrapper = shallow(<NegativeButton
            onClick={onClick} 
            icon={icon}
            size={size}
            theme={theme}
        />);

        // Assert
        expect(wrapper.exists()).toBe(true);        
    })

    test('onClick is called correctly', () => {
        // Arrange
        const onClick = jest.fn();
        const icon = 'close';
        const size = 'small';
        const theme = 'test';
        const wrapper = shallow(<NegativeButton 
            onClick={onClick} 
            icon={icon}
            size={size}
            theme={theme}
        />)

        // Act
        wrapper.find('.negBtn').simulate('click');

        // Assert
        expect(onClick).toHaveBeenCalledTimes(1);
    })

    test('correct styling for button and child icon', () => {
        // Arrange
        const onClick = () => {};
        const icon = 'close';
        const size = 'small';
        const theme = { negativeButton: { color: 'black' } };
        
        // Act
        const wrapper = shallow(<NegativeButton
            onClick={onClick} 
            icon={icon}
            size={size}
            theme={theme}
        />);

        // Assert
        const buttonProps = wrapper.find('.negBtn').get(0).props;
        const iconProps = wrapper.find('.negBtn').get(0).props.children.props;
        expect(buttonProps.className).toEqual('negBtn');
        expect(buttonProps.style).toBe(theme.negativeButton);
        expect(iconProps.className).toEqual('icon');
        expect(iconProps.name).toBe(icon);
        expect(iconProps.size).toBe(size);
    })
})

