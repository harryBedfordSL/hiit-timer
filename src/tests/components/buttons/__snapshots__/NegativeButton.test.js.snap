// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Negative button tests renders without crashing (with snapshot) 1`] = `
<button
  className="negBtn"
  onClick={[Function]}
>
  <i
    aria-hidden="true"
    className="close small icon icon"
    onClick={[Function]}
  />
</button>
`;
